File: lib/old/mm_scraper-scraper/parsers/lib/proxies.js
let fs = require('fs');
let config = require('./../config');
let proxyEnabled = config.proxyEnabled;

let proxy_list = fs.readFileSync(config.proxies).toString().split('\n').map(item => {
  let pair = item.split(':');
  return {
    port: pair[1],
    host: pair[0],
  }
});

console.log(`Loaded ${proxy_list.length} proxies...`);

let proxyObj = {
  list: proxy_list,
  random: function() {
    if (proxyEnabled) {
      let len = proxy_list.length;
      let index = Math.floor(Math.random() * len);

      return proxy_list[index];
    }
  },
  getProxy: function() {
    if (proxyEnabled) {
      let proxy = proxyObj.random();
      return `http://${proxy.host}:${proxy.port}`;
    }
  },
}

module.exports = proxyObj;

File: lib/old/mm_scraper-scraper/scraper/app/proxies.rb
require 'net/http'
require 'openssl'
require 'thread/pool'
require 'http'
require_relative './../config/constants'

class Proxy
  def initialize
    path = File.dirname(__FILE__) + "/../config/valid_proxies.txt"
    @proxies = File.readlines(path)
  end

  def get_proxy
    proxy = random_proxy
    parse_proxy(proxy)
  end

  def random_proxy
    length = @proxies.count
    index = (rand * length).floor
    @proxies[index]
  end

  def parse_proxy(proxy)
    host, port = proxy.split(':')
    {
      :host => host,
      :port => port.to_i
    }
  end
end
